plugins/com.google.cloud.tools.eclipse.appengine.deploy.test/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineDeployInfoTest.java

```
@@ -0,0 +1,70 @@
+                                            + "<appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\">";
+  private static final String PROJECT_ID = "<application>" + TEST_ID + "</application>";
+  private static final String VERSION = "<version>" + TEST_VERSION + "</version>";
+
+  private static final String XML_WITHOUT_PROJECT_ID = XML_PREFIX + XML_SUFFIX;
+  private static final String XML_WITHOUT_VERSION = XML_PREFIX + PROJECT_ID + XML_SUFFIX;
+  private static final String XML_WITH_VERSION_AND_PROJECT_ID = XML_PREFIX + PROJECT_ID + VERSION + XML_SUFFIX;
+
+  @Test
+  public void testParse_noProjectId() throws ParserConfigurationException, SAXException, IOException, CoreException {
+    File xml = createFileWithContent(XML_WITHOUT_PROJECT_ID);
+
+    AppEngineDeployInfo deployInfo = new AppEngineDeployInfo();
+    deployInfo.parse(xml);
+    assertNull(deployInfo.getProjectId());
+  }
+
+  @Test
+  public void testParse_noVersion() throws ParserConfigurationException, SAXException, IOException, CoreException {
+    File xml = createFileWithContent(XML_WITHOUT_VERSION);
+
+    AppEngineDeployInfo deployInfo = new AppEngineDeployInfo();
+    deployInfo.parse(xml);
+    assertThat(deployInfo.getProjectId(), is(TEST_ID));
+    assertNull(deployInfo.getProjectVersion());
+  }
+
+  @Test
+  public void testParse_properXml() throws ParserConfigurationException, SAXException, IOException, CoreException {
+    File xml = createFileWithContent(XML_WITH_VERSION_AND_PROJECT_ID);
+
+    AppEngineDeployInfo deployInfo = new AppEngineDeployInfo();
+    deployInfo.parse(xml);
+    assertThat(deployInfo.getProjectId(), is(TEST_ID));
+    assertThat(deployInfo.getProjectVersion(), is(TEST_VERSION));
+  }
+
+  private File createFileWithContent(String xmlWithoutProjectId) throws IOException {
+    File tempFile = File.createTempFile(getClass().getName(), null);
+    tempFile.deleteOnExit();
+    try (PrintStream stream = new PrintStream(tempFile)) {
```

PrintStream is flaky because it eats exceptions and depends on the system it's running on. Try an OutputStreamWriter connected to a FileOutputStream instead